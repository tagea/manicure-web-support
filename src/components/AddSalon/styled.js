import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import GridList from '@material-ui/core/GridList';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import Button from '@material-ui/core/Button';

export const FormBlockStyled = styled(Grid)`
  padding-top: 10px;
`;

export const DayStyled = styled(Typography)`
  text-transform: capitalize;
  text-align: right;
  &&{
    margin-right: 10px;
    line-height: 46px;
  }
`;

export const BlockPhotosStyled = styled(Grid)`
  overflow: hidden;
  height: 190px;
`;

export const GridListStyled = styled(GridList)`
  &&{
    flex-wrap: nowrap;
  }
  transform: translateZ(0);
`;

export const ButtonAddStyled = styled(Grid)`
  text-align: center;
  position: relative;
  top: calc(90px - 40px / 2);
`;

export const GridListTileBarStyled = styled(GridListTileBar)`
  div{
    flex-grow: 1;
  }
`;

export const InputFileStyled = styled('input')`
  display: none;
`;

export const AddPhotoStyled = styled(Button)`
  &&{
    display: inline-block;
    height: 40px;
    border-radius: 50%;
    min-width: 40px;
    width: 40px;
    padding: 9px;
  }
`;
