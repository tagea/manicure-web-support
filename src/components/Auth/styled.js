import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const PaperStyled = styled(Grid)`
  &&{
    width: 324px;
  }
  border-radius: 4px;
  background-color: ${palette.surface.main};
  padding: 42px 27px;
`;
