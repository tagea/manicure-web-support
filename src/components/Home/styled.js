import styled from 'styled-components';
import { Typography } from '@material-ui/core';

export const TopHeaderStyled = styled(Typography)`
  flex-grow: 1;
`;
