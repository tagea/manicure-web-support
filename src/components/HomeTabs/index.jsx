import React from 'react';
import PropTypes from 'prop-types';
import Tabs from '@material-ui/core/Tabs';
import Grid from '@material-ui/core/Grid';
import { FabButton } from '../../ui/FabButton';
import LogoutSrc from '../../img/logout.svg';

import { ContainerStyled, TabContainerStyled, AppBarStyled, TabStyled, LogOutButtonStyled } from './styled';

class HomeTabs extends React.Component {
  state = {
    value: 0
  };

  handleChange = (event, value) => {
    this.setState({ value });
  };

  render() {
    const { value } = this.state;
    const { titles, children, handleSubmit } = this.props;

    return (
      <ContainerStyled container justify="center">
        <Grid item md={10}>
          <AppBarStyled position="static" color="default">
            <Tabs value={value} onChange={this.handleChange} indicatorColor="primary" textColor="primary">
              {titles.map((title, i) => <TabStyled key={i} label={title} />)}
            </Tabs>
            <LogOutButtonStyled>
              <FabButton label="Logout" src={LogoutSrc} handleClick={handleSubmit} />
            </LogOutButtonStyled>
          </AppBarStyled>
          { children.map((child, i) => (
            value === i && <TabContainerStyled key={i}>{child}</TabContainerStyled>
          )) }
        </Grid>
      </ContainerStyled>
    );
  }
}

HomeTabs.propTypes = {
  titles: PropTypes.arrayOf(PropTypes.string),
  children: PropTypes.arrayOf(PropTypes.object),
  handleSubmit: PropTypes.func
};

HomeTabs.defaultProps = {
  titles: [''],
  children: [{}],
  handleSubmit: () => false
};

export { HomeTabs };
