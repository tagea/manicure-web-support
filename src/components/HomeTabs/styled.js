import styled from 'styled-components';
import Grid from '@material-ui/core/Grid';
import AppBar from '@material-ui/core/AppBar';
import Tab from '@material-ui/core/Tab';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const ContainerStyled = styled(Grid)`
  flex: auto;
  overflow: hidden;
`;

export const TabContainerStyled = styled(Grid)`
  height: 100%;
  margin-top: 30px;
`;

export const AppBarStyled = styled(AppBar)`
  &&{
    box-shadow: none;
    border-radius: 3px;
    position: relative;
    >button{
      top: 4px;
    }
  }
`;

export const TabStyled = styled(Tab)`
  &&{
    font-size: 1.2em;
    padding: 6px 0;
    color: ${palette.text.main};
  }
`;

export const LogOutButtonStyled = styled(Grid)`
  position: absolute;
  right: 20px;
  top: 4px;
`;
