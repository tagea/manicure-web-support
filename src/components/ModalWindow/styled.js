import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';
import DialogTitle from '@material-ui/core/DialogTitle';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const GroupButtonStyled = styled(Typography)`
  button{
    margin: 0 4px;
  }
`;

export const DialogTitleStyled = styled(DialogTitle)`
  border-bottom: 1px solid ${palette.border.light};
  &&{
    padding-bottom: 26px;
    margin-bottom: 30px;
  }
`;
