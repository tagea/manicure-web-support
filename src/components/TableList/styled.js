import styled from 'styled-components';
import Paper from '@material-ui/core/Paper';
import TableCell from '@material-ui/core/TableCell';
import Grid from '@material-ui/core/Grid';

import { Table } from 'react-virtualized';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const PaperStyled = styled(Paper)`
  &&{
    background-color: ${palette.surface.light};
    border-radius: 3px;
    box-shadow: none;
    height: 100%;
  }
`;

export const TableStyled = styled(Table)`
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  box-sizing: border-box;
  div:focus{
      outline: none;
  }
  .rowTable{
    display: flex;
    padding: 30px 25px;
    box-sizing: border-box;
    border-bottom: 1px solid ${palette.border.light};
    border-top: 1px solid transparent;
    transition: all 0.2s ease-in;
    :hover{
      background-color: ${palette.surface.transparent};
      border-bottom: 1px solid ${palette.border.main};
      border-top: 1px solid ${palette.primary.main};
    }
  }
  .ReactVirtualized__Table__headerRow:hover{
    background-color: transparent;
    border-bottom: 1px solid ${palette.border.light};
    border-top: 1px solid transparent;
    
  }
  .headerStyle{
    padding-top: 0;
    padding-bottom: 0;
    line-height: 20px;
    color: ${palette.text.transparent};
  }
`;

export const TableCellStyled = styled(TableCell)`
  &&{
    font-size: 1.1em;
    border-bottom: none;
    padding: 0px;
  }
`;

export const TableHeader = styled(Grid)` 
  &&{
    display:flex;
    justify-content: space-between;
    margin: 0 20px;
    padding-top: 25px;
  }
`;
