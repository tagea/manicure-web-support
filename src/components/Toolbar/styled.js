import styled from 'styled-components';
import AppBar from '@material-ui/core/AppBar';

export const AppBarStyled = styled(AppBar)` 
  &&{
      background-color: transparent;   
      box-shadow: none;                  
      align-items: flex-end;
      justify-content: center;
      padding-top: 10px;
  }
`;
