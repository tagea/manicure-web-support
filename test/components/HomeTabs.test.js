import React from 'react';
import { shallow } from 'enzyme';
import Grid from '@material-ui/core/Grid';
import Tabs from '@material-ui/core/Tabs';
import { HomeTabs } from '../../src/components/HomeTabs';
import { ContainerStyled, AppBarStyled, TabStyled } from '../../src/components/HomeTabs/styled';

it('render HomeTabs component', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent).toMatchSnapshot();
});

it('ContainerStyled should contain "container" property', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(ContainerStyled).prop('container')).toBeTruthy();
});

it('ContainerStyled should contain "justify" property with "center" value', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(ContainerStyled).prop('justify')).toEqual('center');
});

it('Grid should contain "item" property', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(Grid).prop('item')).toBeTruthy();
});

it('Grid should contain "md" property with "10" value', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(Grid).prop('md')).toBeWithinRange(1, 12);
});

it('AppBar should contain "color" property with "default or primary or secondary" value', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(AppBarStyled).prop('color')).toMatch(/primary|secondary|default/);
});

it('AppBar should contain "position" property with "static" value', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(AppBarStyled).prop('position')).toBe('static');
});

it('FabButton should contain label with text', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find('FabButton').prop('label')).not.toBe('');
});

it('FabButton should contain src width text', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find('FabButton').prop('src')).not.toBe('');
});

it('FabButton should have onCLick function', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find('FabButton').prop('handleClick')).toBeInstanceOf(Function);
});

it('Tabs should contain "indicatorColor" property with "primary" value', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(Tabs).prop('indicatorColor')).toBe('primary');
});

it('Tabs should contain "textColor" property with "primary" value', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(Tabs).prop('textColor')).toBe('primary');
});

it('Tabs should have onChange function', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(Tabs).prop('onChange')).toBeInstanceOf(Function);
});

it('Tabs should contain "value" property with initial value "0"', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(Tabs).prop('value')).toBeFalsy();
});

it('Tab should contain "label" property with "" value', () => {
  const HomeTabsComponent = shallow(<HomeTabs />);
  expect(HomeTabsComponent.find(TabStyled).prop('label')).toBe('');
});

expect.extend({
  toBeWithinRange(received, floor, ceiling) {
    const pass = received >= floor && received <= ceiling;
    if (pass) {
      return {
        message: () => `expected ${received} not to be within range ${floor} - ${ceiling}`,
        pass: true
      };
    }
    return {
      message: () => `expected ${received} to be within range ${floor} - ${ceiling}`,
      pass: false
    };
  }
});
