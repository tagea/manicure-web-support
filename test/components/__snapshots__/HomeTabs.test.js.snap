// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render HomeTabs component 1`] = `
<styled__ContainerStyled
  container={true}
  justify="center"
>
  <WithStyles(Grid)
    item={true}
    md={10}
  >
    <styled__AppBarStyled
      color="default"
      position="static"
    >
      <WithStyles(Tabs)
        indicatorColor="primary"
        onChange={[Function]}
        textColor="primary"
        value={0}
      >
        <styled__TabStyled
          key="0"
          label=""
        />
      </WithStyles(Tabs)>
      <FabButton
        handleClick={[Function]}
        label="Logout"
        src="test-file-stub"
      />
    </styled__AppBarStyled>
    <styled__TabContainerStyled
      key="0"
    >
      <Component />
    </styled__TabContainerStyled>
  </WithStyles(Grid)>
</styled__ContainerStyled>
`;
